codex-process-feasibility/src/test/java/de/netzwerk_universitaetsmedizin/codex/processes/feasibility/service/DownloadFeasibilityResourcesTest.java

```diff
@@ -0,0 +1,187 @@
+import static org.highmed.dsf.bpe.ConstantsBase.BPMN_EXECUTION_VARIABLE_TASK;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.StrictStubs.class)
+public class DownloadFeasibilityResourcesTest {
+
+    @Mock
+    private EnhancedFhirWebserviceClientProvider clientProvider;
+
+    @Mock
+    private FhirWebserviceClient webserviceClient;
+
+    @Mock
+    private TaskHelper taskHelper;
+
+    @Mock
+    private DelegateExecution execution;
+
+    @Mock
+    private Task task;
+
+    @InjectMocks
+    private DownloadFeasibilityResources service;
+
+    private Map<String, List<String>> createSearchQueryParts(String measureId) {
+        return Map.of("_id", Collections.singletonList(measureId), "_include",
+                Collections.singletonList("Measure:depends-on"));
+    }
+
+    @Test
+    public void testDoExecute_NoMeasureReference() {
+        when(execution.getVariable(BPMN_EXECUTION_VARIABLE_TASK)).thenReturn(task);
+        when(taskHelper.getFirstInputParameterReferenceValue(task, CODESYSTEM_FEASIBILITY,
+                CODESYSTEM_FEASIBILITY_VALUE_MEASURE_REFERENCE))
+                .thenReturn(Optional.empty());
+
+        assertThrows(RuntimeException.class, () -> service.execute(execution));
+        verify(task).setStatus(Task.TaskStatus.FAILED);
+    }
+
+    @Test
+    public void testDoExecute_BundleWithTooFewResultEntries() {
+        String measureId = "foo";
+        IdType measureRefId = new IdType("Measure/" + measureId);
+        Reference measureRef = new Reference(measureRefId);
+
+        when(execution.getVariable(BPMN_EXECUTION_VARIABLE_TASK))
+                .thenReturn(task);
+        when(taskHelper.getFirstInputParameterReferenceValue(task, CODESYSTEM_FEASIBILITY,
+                CODESYSTEM_FEASIBILITY_VALUE_MEASURE_REFERENCE))
+                .thenReturn(Optional.of(measureRef));
+        when(clientProvider.getWebserviceClient(measureRefId))
+                .thenReturn(webserviceClient);
+        when(webserviceClient.searchWithStrictHandling(Measure.class, createSearchQueryParts(measureId)))
+                .thenReturn(new Bundle());
+
+        assertThrows(RuntimeException.class, () -> service.execute(execution));
+        verify(task).setStatus(Task.TaskStatus.FAILED);
+    }
+
+    @Test
+    public void testDoExecute_FirstBundleEntryIsNoMeasure() {
+        String measureId = "foo";
+        IdType measureRefId = new IdType("Measure/" + measureId);
+        Reference measureRef = new Reference(measureRefId);
+
+        Bundle.BundleEntryComponent patientEntryA = new Bundle.BundleEntryComponent()
+                .setResource(new Patient().setId("foo"));
+
+        Bundle.BundleEntryComponent patientEntryB = new Bundle.BundleEntryComponent()
+                .setResource(new Patient().setId("foo"));
+
+        Bundle measureOnlyBundle = new Bundle()
```

Comment:
This is not a measureOnlyBundle :-) And please use the builder-style pattern to add the entries. This saves many lines of code. I know the HAPI classes are not desinged very well because they are their own builders. But I would use that nevertheless, because it's shorter and I've seen it many times.

Reply:
Made some changes. Hope this fits your requirements.